### PR TITLE
test: Add klog lock error to allow-list

### PR DESCRIPTION
**Pull request by pchaigno**
This error message happened in CI and seems to be a less frequent variation of known klog error messages [1].

1 - https://github.com/cilium/cilium/issues/16402#issuecomment-871155492

### CODE REVIEW

**Comment by pchaigno** [comment]
test-me-please

File: test/helpers/cons.go [hunk 1]
```diff
@@ -241,13 +241,14 @@ const (
...
-	removingInexistantID = "removing identity not added to the identity manager!"        // cf. https://github.com/cilium/cilium/issues/16419
-	failedToListCRDs     = "the server could not find the requested resource"            // cf. https://github.com/cilium/cilium/issues/16425
+	lrpExists          = "local-redirect service exists for frontend"                         // cf. https://github.com/cilium/cilium/issues/16400
+	opCantBeFulfilled  = "Operation cannot be fulfilled on leases.coordination.k8s.io"        // cf. https://github.com/cilium/cilium/issues/16402
+	initLeaderElection = "error initially creating leader election record: leases."           // cf. https://github.com/cilium/cilium/issues/16402#issuecomment-861544964
+	lockDeletedEp      = "lock failed: endpoint is in the process of being removed"           // cf. https://github.com/cilium/cilium/issues/16422
+	globalDataSupport  = "kernel doesn't support global data"                                 // cf. https://github.com/cilium/cilium/issues/16418
+	removeInexistantID = "removing identity not added to the identity manager!"               // cf. https://github.com/cilium/cilium/issues/16419
```
Comment (nbusseneau) [inline test/helpers/cons.go]: Nit: while we're renaming these:
```suggestion
	removeInexistentID = "removing identity not added to the identity manager!"               // cf. https://github.com/cilium/cilium/issues/16419
```

**Review by nbusseneau — APPROVED** [review]

File: test/helpers/cons.go (hunk 1 above)
Comment (pchaigno) [reply test/helpers/cons.go]: :exploding_head:

**Comment by pchaigno** [comment]
Tests are compiling, previous end-to-end test runs were okay, and review is in. Marking as ready and merging.

**Comment by joestringer** [comment]
@pchaigno wait, how do we know that Cilium isn't causing the apiserver connectivity issue that leads to ` http2: client connection lost` ?

**Comment by pchaigno** [comment]
It might. It's not a new error, but we should definitely track it and figure out the root cause.

**Pull request merged**